src/components/WebComponentProject/WebComponentProject.jsx

```diff
@@ -52,9 +52,11 @@ const WebComponentProject = ({
   const [codeHasRun, setCodeHasRun] = useState(codeHasBeenRun);
   const dispatch = useDispatch();
 
-  dispatch(setIsSplitView(outputSplitView));
-  dispatch(setWebComponent(true));
-  dispatch(setIsOutputOnly(outputOnly));
+  useEffect(() => {
+    dispatch(setIsSplitView(outputSplitView));
+    dispatch(setWebComponent(true));
+    dispatch(setIsOutputOnly(outputOnly));
+  }, [outputSplitView, outputOnly, dispatch]);
```

Comment:
I wonder if the dispatch could be skipped sine it's a stable reference

Reply:
We need these dispatches to put the data from the web component attributes into state. But you're right, once it's rendered the first time, these values shouldn't change - the dependencies in the array are just there to keep the linter happy (annoyingly).